lib/matplotlib/animation.py

```diff
@@ -1623,7 +1624,9 @@ def __init__(self, fig, func, frames=None, init_func=None, fargs=None,
         elif callable(frames):
             self._iter_gen = frames
         elif np.iterable(frames):
-            self._iter_gen = lambda: iter(frames)
+            # _iter_gen() must return a new iterator on each call to support
+            # repeat=True. We need to copy since iter() does not ensure this.
+            self._iter_gen = lambda: iter(copy.copy(frames))
```

Comment:
Strictly speaking, copy would only be necessary if frames `not (iter(frames) is frames)`, but I don't think the added complexity is worth trying to save a copy. `frames` is typically not worse than a list of 1000 elements. Copying that is negligable compared to the fact that we have to do a draw step for every frame when rendering the animation.